python/ray/_private/serialization.py

```diff
@@ -379,6 +380,9 @@ def _deserialize_object(self, data, metadata, object_ref):
                 return ActorUnschedulableError(error_info.error_message)
             elif error_type == ErrorType.Value("END_OF_STREAMING_GENERATOR"):
                 return ObjectRefStreamEndOfStreamError()
+            elif error_type == ErrorType.Value("ACTOR_UNAVAILABLE"):
+                error_info = self._deserialize_error_info(data, metadata_fields)
+                return ActorUnavailableError(error_info.error_message)
```

Comment:
Restarting comes in “when sending task request, we find the actor state is restarting so we don’t have a doc peer to send”. Connection break comes in “when waiting for peer response we got a grpc error, and the actor is not known dead by gcs”.

It’s kinda blurry between the two because when you get a grpc issue it’s well possible the actor is dead and restarting.

we can of course distinguish them but I wonder if it’s really useful. How do you see users may want to treat them differently?

Reply:
It's helpful when debugging– suppose the cluster has high latency between nodes and runs applications that restart often (such as those running LLM engines). If we see a spike in `ActorUnavailableErrors`, it would be useful to distinguish between whether they were caused by high latency or by actor restarts.

Reply:
> I don’t think we treat latency. A high latency will just cause the ray.get to wait longer… at least in my test.

Oh in that case, what needs to be wrong with the network for the `RayActorError` to get raise while an actor is still alive?

> In the error message there is a substring RESTARTING. Can we grep it when debugging?

That'll be helpful for debugging after the fact. If we want to build metrics around these failures or do some special handling in the code, we should not be parsing error messages. Can you either (1) include an enum in the exception that describes the cause or (2) split this exception into 2 separate exceptions?